app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -1,92 +1,25 @@
 package com.example.lastfm;
 
-import androidx.appcompat.app.AppCompatActivity;
-import androidx.recyclerview.widget.LinearLayoutManager;
-import androidx.recyclerview.widget.RecyclerView;
-import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
-import android.content.Intent;
-import android.os.Bundle;
-import android.widget.Toast;
-import java.util.ArrayList;
-import java.util.List;
 
-import io.reactivex.SingleObserver;
-import io.reactivex.android.schedulers.AndroidSchedulers;
-import io.reactivex.disposables.Disposable;
-import io.reactivex.schedulers.Schedulers;
+import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
-    private final int NUMBER_OF_ARTISTS = 40;
-    private List<Artist> requestedArtists = new ArrayList<>();
-    private SwipeRefreshLayout mSwipeRefreshLayout;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
 
-    ArtistAdapter artistsAdapter = new ArtistAdapter(this::onArtistClick);
+public class MainActivity extends FragmentActivity {
```

Comment:
It will be better to use AppCompatActivity for both fragments

Reply:
Changed to Appcompatactivity